core/src/io/anuke/mindustry/graphics/BlockRenderer.java

```diff
@@ -269,15 +273,53 @@ void cacheChunk(int cx, int cy, boolean floor){
 		cbatch.end();
 		cache[cx][cy][floor ? 0 : 1] = cbatch.getLastCache();
 	}
-
+	
 	public void clearTiles(){
 		cache = null;
 		createBatch();
 	}
-
+	
 	private void createBatch(){
 		if(cbatch != null)
 			cbatch.dispose();
 		cbatch = new CacheBatch(world.width() * world.height() * 4);
 	}
-}
+	
+	public void drawPreview(Block block, float drawx, float drawy, float rotation, float opacity) {
+		Draw.reset();
+		Draw.alpha(opacity);
```

Comment:
Draw.alpha and Draw.reset both should be moved to the handlePreview function after the multiblock check